### PR TITLE
[wasm] Try to fix then disable unmanaged collator

### PR DESCRIPTION
**Pull request by lewing**
This implements case invariant comparisons in the unmanaged internal_index path

### CODE REVIEW

**Comment by lewing** [comment]
This mode looks a lot more like GlobalizationMode.Invariant.  Should we throw a 
`
	<type fullname=" System.Globalization.GlobalizationMode">
	  <method signature="System.Boolean get_Invariant()" body="stub" value="false"/>
	</type>
` 
into the linker substitutions?

**Comment by lewing** [comment]
This contains #19571 and 
fixes #19034 #18978

File: mcs/class/corlib/Test/System/StringTest.cs [hunk 1]
```diff
@@ -3221,6 +3221,8 @@ public void Replace()
 
 		// Test replacing null characters (bug #67395)
 		Assert.AreEqual ("is this ok ?", "is \0 ok ?".Replace ("\0", "this"), "should not strip content after nullchar");
+
+		Assert.AreEqual ("CBC", "ABC".Replace ("a", "C", StringComparison.InvariantCultureIgnoreCase));
```
Comment (pranavkm) [inline mcs/class/corlib/Test/System/StringTest.cs:3225]: https://stackoverflow.com/a/20085219 has a nicer test case.

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
This was very confusing until I dug a little deeper

```
0 = internal_index (BBB, 0, 3, a, IgnoreCase, True)
0 = internal_index (AAA, 0, 3, b, IgnoreCase, True)
1 = internal_index (ABC, 0, 3, a, IgnoreCase, True)
```

So something appears to be very wrong with the g_unichar_to_lower tables we have included

**Comment by kjpou1** [comment]
@monojenkins build Linux WebAssembly

**Comment by kjpou1** [comment]
@monojenkins build Linux x64 mcs

**Comment by kjpou1** [comment]
```
        var s1 = "Strasse";
        var s2 = "Straße";

        var s1e = s1.Equals(s2, StringComparison.Ordinal);           //false
        var s1e2 = s1.Equals(s2, StringComparison.InvariantCulture);  //true
```

output: 

```
False
True
```

File: mcs/class/corlib/Test/System/StringTest.cs (hunk 1 above)
Comment (kjpou1) [reply mcs/class/corlib/Test/System/StringTest.cs:3225]: Reproduced below

**Comment by lewing** [comment]
After further review and some discussion with @marek-safar  it looks like using the managed collator is preferable even if the collation tables are completely removed.  This version of the patch would allow something like:

```
<linker>
  <assembly fullname="mscorlib">
	<resource name="collation.cjkCHS.bin" action="remove"/>
	<resource name="collation.cjkCHT.bin" action="remove"/>
	<resource name="collation.cjkJA.bin" action="remove"/>
	<resource name="collation.cjkKO.bin" action="remove"/>
	<resource name="collation.cjkKOlv2.bin" action="remove"/>
	<!--
	<resource name="collation.core.bin" action="remove"/>
	<resource name="collation.tailoring.bin" action="remove"/>
	-->
	
	<type fullname="System.Globalization.CompareInfo">
	  <method signature="System.Boolean get_InoreCaseNotSupported()" body="stub" value="true"/>
	</type>

	<type fullname="System.Globalization.CompareInfo">
	  <method signature="System.Boolean get_UseManagedCollation()" body="stub" value="false"/>
	</type>
  </assembly>
</linker>
```

In the substitutions file which would would reduce the overhead of the collator substantially since the overwhelming majority of the size increase from the managed collator comes from the collation resources.

File: mcs/class/corlib/ReferenceSources/CompareInfo.cs [hunk 2]
```diff
@@ -83,6 +83,12 @@ partial class CompareInfo
 			}
 		}
 
+		static bool InoreCaseNotSupported {
```
Comment (pranavkm) [inline mcs/class/corlib/ReferenceSources/CompareInfo.cs:86]:
```suggestion
		static bool IgnoreCaseNotSupported {
```

Comment (lewing) [reply mcs/class/corlib/ReferenceSources/CompareInfo.cs]: At least I was consistent in usage. :)

**Comment by lewing** [comment]
The core and tailoring resources could potentially be removed as well.  It comes down to a choice on what the reduced mode should support.   Additionally if the goal is for the default experience to match traditionally behavior this should be opt in not opt out.

**Comment by lewing** [comment]
These are the sizes of the resource files:
```
 55813 collation.cjkCHS.bin
 44549 collation.cjkCHT.bin
 44549 collation.cjkJA.bin
 44549 collation.cjkKO.bin
 22273 collation.cjkKOlv2.bin
118901 collation.core.bin
  6724 collation.tailoring.bin
```

**Comment by lewing** [comment]
@monojenkins build failed

**Comment by lewing** [comment]
I've removed the unmanged collator https://github.com/lewing/mono/tree/unmanaged-collator-ignore-case in case they need to be revisited

**Comment by lewing** [comment]
@monojenkins backport 3.2-wasm

**Pull request closed without merging**